tetragon: add uprobe override action (#4173)
cilium/tetragon: docs/content/en/docs/concepts/tracing-policy/selectors.md

ensure they properly follow the [Error Injectable Functions](https://docs.kernel.org/fault-injection/fault-injection.html#error-injectable-functions) guide.
{{< /caution >}}

#### Override action for uprobe

> **BEWARE, HERE BE DRAGONS!!! USE WITH CAUTION COULD EASILY CRASH TRACED APPLICATION**

Reviewer: ```suggestion
{{< warning >}}
Beware, here be dragons!!! Use with caution, it could easily crash traced application.
{{< /warning >}}
```

Note that `Override` action can be used only on `x86_64` architecture.

Similar to kprobe the uprobe `Override` action allows to modify the return value
of user space call with `argError` argument, like:

```yaml
uprobes:
- path: "test"
  symbols:
  - "func"
  selectors:
  - matchActions:
    - action: Override
      argError: 123
```

Note that it can be successfully used only when following conditions are met:
- uprobe is attached to the beggining of the user space function
- user space function is called via `call` instruction

> - user space function returns `int` type

Reviewer: ```suggestion
- uprobe is attached to the beggining of the user space function;
- user space function is called via `call` instruction;
- user space function returns `int` type.
```

It's possible to override specific registers with arbitrary value with `argRegs`
argument, like:

```yaml
uprobes:
- path: "test"
  symbols:
  - "func"
  selectors:
  - matchActions:
    - action: Override
      argRegs:
      - "rax=11"
      - "rbp=(%rsp)"
      - "rip=8(%rsp)"
      - "rsp=8%rsp"
```

The `argRegs` argument is array of strings where each string start with destination
register name followed by `=` and assignment expression, which can be one of the
following types:

- constant `rax=11`
- register `rbp=(%rsp)`
- register plus offset `rip=8(%rsp)`
- dereference of register plus offset `rsp=8%rsp`

> Note that this interface is likely to be changed in the future.

Reviewer: ```suggestion
{{< caution >}}
This interface is likely to be changed in the future.
{{< /warning >}}
```
or
```suggestion
{{< note >}}
This interface is likely to be changed in the future.
{{< /note >}}
```
